### PR TITLE
Add dynamic content to tooltipinfo in presentation selection

### LINKED ISSUE #213: Feat: Update tooltipinfo for each presentation type
This  one:

tooltipinfo-table

### PR DESCRIPTION
**Pull request by oysor**
## Issues closed
#213

### CODE REVIEW

**Comment by Log234** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817&bid=19980878)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 1]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                return {
+                    text: "Her kan du velge fylke og se en fullstendig oversikt for valgt fylke.",
+                    url: "#Enkeltfylke"
+                };
+            }
+            default: {
+                return {
+                    text: "Trykk på ikonet for å lese om oversikten.",
+                    url: "#Visning"
+                };;
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:73]: Codacy found an issue: [Unnecessary semicolon.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 2]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                    url: "#Utjevningsmandater."
+                };
+            }
+            case 'TABLE_REMAINDER_QUOTIENTS': {
+                return {
+                    text: "Her ser du en oversikt over restkvotienter.",
+                    url: "#Restkvotienter"
+                };
+            }
+            case 'SEAT_DISTRIBUTION': {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:57]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 3]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                return {
+                    text: "Her ser du fordelingen av mandater på hvert fylke.",
+                    url: "#Fylkesfordeling%20av%20mandater"
+                };
+            }
+            case 'TABLE_SINGLE_COUNTY': {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:63]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 4]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                    url: "#Landsoversikt"
+                };
+            }
+            case 'TABLE_DISTRICT_OVERVIEW': {
+                return {
+                    text: "Her kan du se valgresultatet for hvert fylke.",
+                    url: "#Fylkesoversikt"
+                };
+            }
+            case 'TABLE_LEVELLING_SEATS_OVERVIEW': {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:45]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 5]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                return {
+                    text: "Her kan du se fra hvilke fylker partiene har fått utjevningsmandater fra.",
+                    url: "#Utjevningsmandater."
+                };
+            }
+            case 'TABLE_REMAINDER_QUOTIENTS': {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:51]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 6]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
         });
     };
 
+    selectTooltipContent(table: string | undefined) {
+        switch (table) {
+            case 'TABLE_ELECTION_OVERVIEW': {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:33]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 7]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                return {
+                    text: "Her kan du se valgresultatet på landsbasis. Trykk på meg for mer informasjon!",
+                    url: "#Landsoversikt"
+                };
+            }
+            case 'TABLE_DISTRICT_OVERVIEW': {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:39]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6103817)

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 8]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
         });
     };
 
+    selectTooltipContent(table: string | undefined) {
+        switch (table) {
+            case "TABLE_ELECTION_OVERVIEW": {
+                return {
+                    text: "Her kan du se valgresultatet på landsbasis. Trykk på meg for mer informasjon!",
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:35]: I would suggest we remove the "Trykk på meg for mer informasjon!" as this is already mentioned in the tutorial.

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 9]
```diff
@@ -28,13 +28,61 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                };
+            }
+            case "TABLE_SINGLE_COUNTY": {
+                return {
+                    text: "Her kan du velge fylke og se en fullstendig oversikt for valgt fylke.",
+                    url: "#Enkeltfylke"
+                };
+            }
+            default: {
+                return {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:70]: The default case should use the checkExhaustively function to ensure we never reach the default case.

**Review by Log234 — CHANGES_REQUESTED** [review]
Two minor tweaks, but otherwise a great PR!

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 10]
```diff
@@ -28,13 +29,58 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                    url: "#Utjevningsmandater."
+                };
+            }
+            case PresentationType.SeatDistribution: {
+                return {
+                    text: "Her ser du en oversikt over restkvotienter.",
+                    url: "#Restkvotienter"
+                };
+            }
+            case PresentationType.RemainderQuotients: {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:58]: You seem to have swapped the texts for RemainderQuotients and SeatDistribution.

File: src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx [hunk 11]
```diff
@@ -28,13 +29,58 @@ export class PresentationSelection extends React.Component<PresentationSelection
...
+                    url: "#Fylkesoversikt"
+                };
+            }
+            case PresentationType.LevellingSeats: {
+                return {
+                    text: "Her ser du fra hvilke fylker partiene har fått utjevningsmandater fra.",
+                    url: "#Utjevningsmandater."
+                };
+            }
+            case PresentationType.SeatDistribution: {
```
Comment (Log234) [inline src/Layout/PresentationMenu/PresentationSelection/PresentationSelection.tsx:52]: You seem to have swapped the texts for SeatDistribution and RemainderQuotients.

**Review by Log234 — CHANGES_REQUESTED** [review]
There are two texts that appear to have been swapped.

**Comment by oysor** [comment]
The tooltip text could probably be better.

**Comment by Log234** [comment]
@oysor Would you mind adding a tooltip for the new table before we merge?

**Review by Log234 — APPROVED** [review]
Looks good to me!

**Pull request merged**